sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AcsCallEndedEventData.java

```diff
@@ -0,0 +1,229 @@
+// Code generated by Microsoft (R) AutoRest Code Generator.
+package com.azure.messaging.eventgrid.systemevents;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import java.io.IOException;
+import java.time.Duration;
+
+/**
+ * Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallEnded event.
+ */
+@Fluent
+public final class AcsCallEndedEventData extends AcsCallingEventProperties {
+
+    /*
+     * The communication identifier of the user who was disconnected
+     */
+    private AcsCallEndedByProperties endedBy;
+
+    /*
+     * The reason for ending the call.
+     */
+    private AcsCallEndReasonProperties reason;
+
+    /*
+     * Duration of the call in seconds.
+     */
+    private Float callDurationInSeconds;
+
+    /**
+     * Creates an instance of AcsCallEndedEventData class.
+     */
+    public AcsCallEndedEventData() {
+    }
+
+    /**
+     * Get the endedBy property: The communication identifier of the user who was disconnected.
+     *
+     * @return the endedBy value.
+     */
+    public AcsCallEndedByProperties getEndedBy() {
+        return this.endedBy;
+    }
+
+    /**
+     * Set the endedBy property: The communication identifier of the user who was disconnected.
+     *
+     * @param endedBy the endedBy value to set.
+     * @return the AcsCallEndedEventData object itself.
+     */
+    public AcsCallEndedEventData setEndedBy(AcsCallEndedByProperties endedBy) {
+        this.endedBy = endedBy;
+        return this;
+    }
+
+    /**
+     * Get the reason property: The reason for ending the call.
+     *
+     * @return the reason value.
+     */
+    public AcsCallEndReasonProperties getReason() {
+        return this.reason;
+    }
+
+    /**
+     * Set the reason property: The reason for ending the call.
+     *
+     * @param reason the reason value to set.
+     * @return the AcsCallEndedEventData object itself.
+     */
+    public AcsCallEndedEventData setReason(AcsCallEndReasonProperties reason) {
+        this.reason = reason;
+        return this;
+    }
+
+    /**
+     * Get the callDuration property:Duration of the call in seconds.
+     *
+     * @return the callDuration value.
+     */
+    public Duration getCallDuration() {
+        if (this.callDurationInSeconds != null) {
+            return Duration.ofNanos((long) (this.callDurationInSeconds * 1000_000_000L));
+        }
+        return null;
+    }
+
+    /**
+     * Set the callDuration property: Duration of the call in seconds.
+     *
+     * @param callDuration the callDuration value to set.
+     * @return the AcsCallEndedEventData object itself.
+     */
+    public AcsCallEndedEventData setCallDuration(Duration callDuration) {
+        if (callDuration != null) {
+            this.callDurationInSeconds = callDuration.toNanos() / 1_000_000_000f;
+        }
+        return null;
```

Comment:
We should return `this` here. Also, if the callDuration is null and `callDurationInSeconds was previously set to a non-null value. We should clear it out.